### ISSUE TITLE
MSVC 2022 compilation error in 3.6.0

### ISSUE THREAD

**Issue by lhog** [taskflow/taskflow#510]
https://gist.github.com/lhog/fd82bf5e36cea94ef50b1701a2c7ba53

causes error:
```
error C4146: unary minus operator applied to unsigned type, result still unsigned
..... taskflow\algorithm\sort.hpp	42
```

Basically it dislikes `-cacheline_size` in https://github.com/taskflow/taskflow/blob/v3.6.0/taskflow/algorithm/sort.hpp#L42

I believe it can be replaced with:
`ip = (ip + cacheline_size - 1) & ~(cacheline_size - 1);`

But I'm not entirely sure.

**Comment by tsung-wei-huang** [comment]
@lhog Yes, you are correct with `~(cacheline_size - 1)`. This has been fixed in the current master branch.
